Smartpad-iOS/Smartpad-iOS/Controller/DragPanGestureRecognizer.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  DragPanGestureRecognizer.swift
+//  Smartpad-iOS
+//
+//  Created by Hudson Shykowski on 2022-03-22.
+//
+
+import Foundation
+import UIKit
+
+// Portions of this code follow the apple documentation:
+// https://developer.apple.com/documentation/uikit/touches_presses_and_gestures/implementing_a_custom_gesture_recognizer/implementing_a_discrete_gesture_recognizer
+//
+
+// Custom gesture for longtouch then drag
+// 1. The user must long touch for 1s
+// 2. The user must pan
+class UIDragPanGestureRecognizer:
+    UIGestureRecognizer {
+    /* Geture states */
+    private enum DragPanPhases {
+        case notStarted /* State when we have touched down but not long touched */
+        case started /* State when we have long touched but not yet panned */
+        case changed /* State when we first started panning */
+    }
+
+    /* Current drag pan state */
+    private var currentPhase = DragPanPhases.notStarted
+
+    /* Work to be run asynchronously after some time (for detecting long press) */
+    // https://stackoverflow.com/questions/44633729/stop-a-dispatchqueue-that-is-running-on-the-main-thread
+    private var work: DispatchWorkItem?
+
+    /* Controls length of required time for long touching */
+    private let longTouchTime = 1
+
+    /* Initial point we touched */
+    var initialPos: CGPoint = CGPoint.zero
+
+    /* Translation from the initial point */
+    var translation: CGPoint = CGPoint.zero
+
+    /* The touch we are tracking */
+    var trackedTouch : UITouch? = nil
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesBegan(touches, with: event)
+
+        /* Only recognizer 1 finger touches */
+        if touches.count != 1 {
+            self.state = .failed
+//            print("Failed, too many touches")
+        }
+        else {
+            // Capture the first touch and store some information about it.
+            if self.trackedTouch == nil {
+                self.trackedTouch = touches.first
+                self.initialPos = (self.trackedTouch?.location(in: self.view))!
+            }
+            else {
+               // Ignore all but the first touch.
+               for touch in touches {
+                  if touch != self.trackedTouch {
+                     self.ignore(touch, for: event)
+                  }
+               }
+            }
+
+            work = DispatchWorkItem(block: {
+                self.currentPhase = .started
+                // TODO: Can we propagate this event to the listener?
+                // We can play a haptic when the long press is "confirmed"
+//                print("Long touch success")
```

Comment:
Once haptic PR is merged I will take a look at playing some haptic feedback once the long press has completed